src/frontend/src/utils/status.ts

```diff
@@ -73,275 +91,310 @@ const defaultFinalReviewModal: Modal = {
+    MPO_REVIEW: {
+      title: 'MPO Review',
+      class: 'statusBlock__MPOReview',
+      modal: defaultMPOReviewModal,
+      Privileges: {
+        MPO: {
+          banner: BannerText.MPOReviewStatusCalloutText.MPO.en,
+          changeStatus: [
+            {
+              status: 'INCOMPLETE',
+              modal: checkReviewStatusDelegate(pia)
+                ? resetReviewIncompleteModal
+                : defaultIncompleteModal,
+            },
+            {
+              status: 'EDIT_IN_PROGRESS',
+              modal: checkReviewStatusDelegate(pia)
+                ? resetReviewEditInPRogressModal
+                : defaultEditInProgressModal,
+            },
+            {
+              status: 'CPO_REVIEW',
+              modal: defaultCPOReviewModal,
+            },
+          ],
+        },
+        CPO: {
+          changeStatus: [
+            {
+              status: 'CPO_REVIEW',
+              modal: defaultCPOReviewModal,
+            },
+            {
+              status: 'INCOMPLETE',
+              modal: checkReviewStatusDelegate(pia)
+                ? resetReviewIncompleteModal
+                : defaultIncompleteModal,
+            },
+            {
+              status: 'EDIT_IN_PROGRESS',
+              modal: checkReviewStatusDelegate(pia)
+                ? resetReviewEditInPRogressModal
+                : defaultEditInProgressModal,
+            },
+          ],
+        },
       },
     },
-  },
-  INCOMPLETE: {
-    title: 'Incomplete',
-    class: 'statusBlock__incomplete',
-    banner: BannerText.InCompleteStatusCalloutText.Drafter.en,
-    modal: defaultIncompleteModal,
-    Privileges: {
-      MPO: {
-        banner: BannerText.InCompleteStatusCalloutText.Drafter.en, //incomplete only have one banner for both mpo and drafter
-        changeStatus: [
-          {
-            status: 'EDIT_IN_PROGRESS',
-            modal: defaultEditInProgressModal,
-          },
-          {
-            status: 'CPO_REVIEW',
-            modal: defaultCPOReviewModal,
-          },
-        ],
-      },
-      CPO: {
-        changeStatus: [
-          {
-            status: 'MPO_REVIEW',
-            modal: defaultMPOReviewModal,
-          },
-          {
-            status: 'EDIT_IN_PROGRESS',
-            modal: defaultEditInProgressModal,
-          },
-          {
-            status: 'CPO_REVIEW',
-            modal: defaultCPOReviewModal,
-          },
-        ],
+    INCOMPLETE: {
+      title: 'Incomplete',
+      class: 'statusBlock__incomplete',
+      banner: BannerText.InCompleteStatusCalloutText.Drafter.en,
+      modal: defaultIncompleteModal,
+      Privileges: {
+        MPO: {
+          banner: BannerText.InCompleteStatusCalloutText.Drafter.en, //incomplete only have one banner for both mpo and drafter
+          changeStatus: [
+            {
+              status: 'EDIT_IN_PROGRESS',
+              modal: checkReviewStatusDelegate(pia)
+                ? resetReviewEditInPRogressModal
```

Comment:
changing from INCOMPLETE to EDIT_IN_PROGRESS shouldn't require the unlock modal.

Reply:
correct. The check should fail as the review should be empty. Having the check should not cause any harm unless the logic is broken somewhere else.